streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -266,7 +266,8 @@ public void registerStore(final StateStore store, final StateRestoreCallback sta
         }
 
         if (stores.containsKey(storeName)) {
-            throw new IllegalArgumentException(format("%sStore %s has already been registered.", logPrefix, storeName));
+            log.warn("State Store {} has already been registered, which could be due to a half-way registration" +
+                "in the previous round", storeName);
```

Comment:
nit: we could make the warn log entry more clear that we did not override the registered the store, e.g. "Skipped registering state store {} since it has already existed in the state manager, ..."